### PR TITLE
Bluetooth: host: Don't push the TS flag on `buf` itself

### LINKED ISSUE #76687: Bluetooth: host: Off-by-one error in MTU calculation leads to ISO fragmentation
There is an off-by-one error when calculating ISO MTU size.
This leads to the `conn.c` layer fragmenting ISO SDUs with the final fragment being only 1 byte.

This is sub-optimal.

To reproduce: run this test: https://github.com/jori-nordic/zephyr/commit/9fc1afedfa53d2923dbb3fe480996330b7086efc

### PR DESCRIPTION
**Pull request by jori-nordic**
When I did that, I messed up the MTU calculations.
Classic off-by-one error :/

Instead, infer the presence of the timestamp by inspecting the amount of
headroom that the buffer has. This works because we always reserve
enough memory to push the timestamp, but not always push a timestamp on
the buffer.

This method is slightly uglier IMO, but eases MTU confusion, and doesn't
rely on user_data.

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/76687

### CODE REVIEW

**Comment by jori-nordic** [comment]
CC @ivaniushkov

File: tests/bsim/bluetooth/host/iso/frag/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  src/broadcaster.c
+  src/main.c
+)
+
+zephyr_include_directories(
+  ${BSIM_COMPONENTS_PATH}/libUtilv1/src/
+  ${BSIM_COMPONENTS_PATH}/libPhyComv1/src/
+)
+
+# [Hacker voice] "I'm in.."
```
Comment (aescolar) [inline tests/bsim/bluetooth/host/iso/frag/CMakeLists.txt:22]: ..

File: tests/bsim/bluetooth/host/iso/frag/src/broadcaster.c [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+
+	buf = net_buf_alloc(&tx_pool, K_NO_WAIT);
+	TEST_ASSERT(buf != NULL, "Failed to allocate buffer");
+
+	net_buf_reserve(buf, BT_ISO_CHAN_SEND_RESERVE);
+
+	net_buf_add_le32(buf, 0xdeadbeef);
+	net_buf_add_u8(buf, 0x11);
+
+	LOG_INF("Sending SDU with%s timestamp (headroom %d)", ts ? "" : "out", net_buf_headroom(buf));
```
Comment (aescolar) [inline tests/bsim/bluetooth/host/iso/frag/src/broadcaster.c:46]: You can run locally 
`scripts/ci/check_compliance.py -c main..HEAD`

File: subsys/bluetooth/host/conn.c [hunk 3]
```diff
@@ -530,6 +530,11 @@ static int send_acl(struct bt_conn *conn, struct net_buf *buf, uint8_t flags)
 	return bt_send(buf);
 }
 
+static enum bt_iso_timestamp contains_iso_timestamp(struct net_buf *buf)
+{
+	return net_buf_headroom(buf) == BT_BUF_ISO_SIZE(0) - sizeof(struct bt_hci_iso_sdu_ts_hdr);
```
Comment (jhedberg) [inline subsys/bluetooth/host/conn.c:535]: You're essentially implicitly converting `bool` to `enum bt_iso_timestamp` here. I think it'd be cleaner to do that conversion explicitly in the function and return an actual enum value.

File: tests/bsim/bluetooth/host/iso/frag/src/broadcaster.c (hunk 2 above)
Comment (jori-nordic) [reply tests/bsim/bluetooth/host/iso/frag/src/broadcaster.c:46]: Yeah I know. I only do it if I have one failure.

File: tests/bsim/bluetooth/host/iso/frag/CMakeLists.txt (hunk 1 above)
Comment (jori-nordic) [reply tests/bsim/bluetooth/host/iso/frag/CMakeLists.txt:22]: Cmon, this hurts nobody. Linker wrapping is wiretapping

**Review by jhedberg — DISMISSED** [review]

**Review by alwa-nordic — APPROVED** [review]

**Comment by rugeGerritsen** [comment]
Can you update the commit title to make it clear that this commit fixes a bug?

**Comment by rugeGerritsen** [comment]
> We don't usually tag the commit titles with [bug] or [fix]. I can add the label to the PR if you want.

I meant something like
"Bluetooth: Host: Fix sending ISO data with timestamp"

**Review by jhedberg — APPROVED** [review]

**Review by Yedfree — APPROVED** [review]

**Pull request merged**